stdlib/public/core/String.swift

```diff
@@ -849,14 +849,29 @@ extension String {
       }
     }
 
-    var result = ""
-    result.reserveCapacity(utf8.count)
+    var buffer: [UInt8] = []
+    buffer.reserveCapacity(utf8.count)
 
     for scalar in unicodeScalars {
-      result += scalar.properties.lowercaseMapping
+      buffer.withUnsafeMutableBytes { buffer in
```

Comment:
Is there a difference in calling `withUnsafeMutableBytes` inside or outside the loop? I'm just asking because I have experience perf impact of such calls that could be outside. Maybe it doesn't make much difference here since is always a small number of iterations, but asking just for curiosity :)